feat(kno-7297): update MS Teams integration docs (#680)
knocklabs/docs: content/integrations/chat/microsoft-teams/sending-a-message-to-channels.mdx

section: Integrations > Microsoft Teams
layout: integrations
---

In this guide, we'll cover how to update a multi-tenant Microsoft Teams bot to send messages to channels in Microsoft Teams using Knock. It assumes that you have already created a Microsoft Teams channel in Knock as outlined in the [Microsoft Teams integration overview](/integrations/chat/microsoft-teams/overview) guide.

Here's what we'll cover in this guide:

- Modeling a multi-tenant application in Knock using `Tenants`
- Adding required scopes to your Microsoft Teams app's manifest
- Setting required Tenant and Object channel data when a Microsoft Teams bot is installed in a channel
- Triggering a workflow with an object recipient to send a message to a Microsoft Teams channel

## Prerequisites

Make sure your bot has been registered with <a href="https://dev.botframework.com/" target="_blank">Microsoft Bot Framework</a> and is deployed to Azure. Knock does not manage deploying and configuring your bot. To set up Knock to send notifications as your bot, see [How to connect to Teams with Knock](/integrations/chat/microsoft-teams/overview#how-to-connect-to-teams-with-knock).

## Key concepts

> There are two key concepts you'll see throughout the following docs that are foundational to how Knock's Microsoft Teams integration works, but might not be used in every implementation of Knock: tenants and objects.

Reviewer: ```suggestion
There are two key concepts you'll see throughout the following docs that are foundational to how Knock's Microsoft Teams integration works: tenants and objects.
```
same suggestion as before, I don't think that caveat is giving any useful info and might be unnecessary to say

If you already use Knock's tenant concept to power other 'account-based' features, you likely create tenants in Knock when an account or organization is created in your application.

<Callout
  emoji="💡"
  text={
    <>
      <strong>Note: </strong>
      Our best-practice recommendation is that tenants in Knock should map one-to-one
      to whatever abstraction you use to model accounts, organizations, or workspaces.
      You can think of tenants as the top-level container within your data model
      that you use to power multi-tenancy in your application.
    </>
  }
/>

### About objects

[Objects](/concepts/objects) in Knock are flexible abstractions meant to map to a resource in your system. Each individual object in Knock exists within a `collection` and requires an `id` unique to that collection.

> In the context of Knock's Microsoft Teams integration, objects serve two purposes. First, they store the Microsoft Teams channel or channels you want to notify. Second, they act as the recipient of the workflow that sends a message to Microsoft Teams.

Reviewer: ```suggestion
In the context of Knock's Microsoft Teams integration, objects serve two purposes. First, they store the Microsoft Teams channel or channels you want to notify. Second, they act as the recipient of the workflow used to send a message to Microsoft Teams.
```

<Callout
  emoji="💡"
  text={
    <>
      <strong>Note: </strong>
      You can think of collections as the different tables within your database that
      represent resources in your application. Objects are the rows within those
      tables.
    </>
  }
/>

#### Example

> Let's say we're building an example source control application like GitHub, where teams can collaborate and share code repositories. In this context, each GitHub organization would map to a tenant in Knock, and each repository would become an object inside of a `repositories` collection.

Reviewer: ```suggestion
Let's say we're building a source control application like GitHub, where teams can collaborate and share code repositories. In this context, each GitHub organization would map to a tenant in Knock, and each repository would become an object inside of a `repositories` collection.
```
